pkg/resourcegenerator/github/image_pull_secret.go

```diff
@@ -0,0 +1,59 @@
+package github
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/kartverket/skiperator/pkg/reconciliation"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type imagePullSecret struct {
+	payload []byte
+}
+
+func NewImagePullSecret(token, registry string) (*imagePullSecret, error) {
+	cfg := dockerConfigJson{}
+	cfg.Auths = make(map[string]dockerConfigAuth, 1)
+	auth := dockerConfigAuth{}
+	auth.Auth = token
+	cfg.Auths[registry] = auth
+
+	var buf bytes.Buffer
+	enc := json.NewEncoder(&buf)
+	err := enc.Encode(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &imagePullSecret{payload: buf.Bytes()}, nil
+}
+
+func (ips *imagePullSecret) Generate(r reconciliation.Reconciliation) error {
+	if r.GetType() != reconciliation.NamespaceType {
+		return fmt.Errorf("image pull secret only supports namespace type")
+	}
+	secret := corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: r.GetSKIPObject().GetName(), Name: "github-auth"}}
```

Comment:
```suggestion
	secret := corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: r.GetSKIPObject().GetNamespace(), Name: "github-auth"}}
```

Reply:
This is correct already, because the SKIPObject here is a namespace. A namespace object has an empty string as value for GetNamespace(). I will add a comment to clarify